substrate/frame/balances/src/impl_currency.rs

```diff
@@ -132,6 +132,12 @@ mod imbalances {
 		}
 	}
 
+	impl<T: Config<I>, I: 'static> TryMerge for PositiveImbalance<T, I> {
+		fn try_merge(self, other: Self) -> Result<Self, (Self, Self)> {
+			Ok(self.merge(other))
```

Comment:
I think conceptually it makes more sense to use `try_merge` from `merge` and just ignore the return value. Then try-merge would fail on overflow instead of saturating.

Reply:
Can you explain why? It feels less correct to me to ignore an error from `try_merge` within `merge`. In the scope of `merge` we wont know for sure, that the `try_merge` wont return an error.
We probably fine with saturation since we do not expect it to overflow the max supply type. It may happened only if you create imbalances in some hacky way, and do not use fungibles api.

> Then try-merge would fail on overflow instead of saturating. 

To handle this we would need a separate impl for `try_merge`, since `merge` is infallible

Reply:
The difference is that we explicitly ignore the error - instead of `merge` pretending to be infallible. We already have messed up TI on production, so i think saturation could happen.  
Anyway, i dont care so much in this case.